pkg/apicast/apicast_option_provider.go

```diff
@@ -122,6 +122,15 @@ func (a *APIcastOptionsProvider) GetApicastOptions() (*APIcastOptions, error) {
 	a.APIcastOptions.Workers = a.APIcastCR.Spec.Workers
 	a.APIcastOptions.Timezone = a.APIcastCR.Spec.Timezone
 
+	for _, customPolicySpec := range a.APIcastCR.Spec.CustomPolicies {
```

Comment:
I'm wondering whether we should validate each secret first, when processing the options?
In the case of HTTPsCertificate we check that it exists beforehand(see `getHTTPSCertificateSecret` method). Maybe check that the specific required fields in the secret are there too?

What do you think?

Reply:
yes! additional check would be good

Reply:
implemented